python/graphstorm/model/edge_gnn.py

```diff
@@ -199,19 +200,16 @@ def edge_mini_batch_gnn_predict(model, loader, return_proba=True, return_label=F
         for iter_l in range(max_num_batch):
             iter_start = time.time()
             tmp_keys = []
+            blocks = None
+            input_nodes = {}
             if iter_l < len_dataloader:
                 input_nodes, target_edge_graph, blocks = next(dataloader_iter)
                 if not isinstance(input_nodes, dict):
                     assert len(g.ntypes) == 1
                     input_nodes = {g.ntypes[0]: input_nodes}
+            if is_wholegraph():
                 tmp_keys = [ntype for ntype in g.ntypes if ntype not in input_nodes]
```

Comment:
Why we still need the tmp_keys here?

Reply:
To remove the additional ntypes before calling the `predict` function in line 230